modules/Bio/EnsEMBL/BaseAlignFeature.pm

```diff
@@ -1009,7 +1012,303 @@ sub _query_unit {
+      if($mdtag =~/^\d+$/){
+        return [$mdtag];
+      }else{
+        my @char_arr = split('',$mdtag);
+        for(my $i=0; $i< scalar(@char_arr); $i++){
+        my $char = $char_arr[$i];
+
+        #determine the type and next type
+        my $type = $self->_get_mdz_chunk_type($char);
+        my $next_char = $char_arr[++$i];
+        my $next_type = $self->_get_mdz_chunk_type( $next_char);
+
+        my $chunk = $char;
+
+        #group chars in mdz based on the type, break if it differenc and push to chunks
+        while($type eq $next_type and $i <= $#char_arr){
+            $chunk = $chunk . $next_char;
+            $next_char = $char_arr[++$i];
+            $next_type = $self->_get_mdz_chunk_type($next_char);
+           }
+
+        if($i == $#char_arr){
+          if($type eq $next_type){
+           $chunk = $chunk . $next_char;
+           push(@chunks, $chunk);
+          }else{
+           push(@chunks, $chunk);
+          }
+        }else{
+          push(@chunks, $chunk);
+        }
+        --$i;
+        }
+      }
+
+    }
+
+  return \@chunks;
+}
+
+=head2 _get_mdz_alignment_length
+
+  Arg [1]    : array of strings
+  Description: calculate the alignment length from the given chunks
+  Returntype : array of strings
+  Exceptions : none
+  Caller     : internal
+  Status     : Stable
+
+=cut
+
+sub _get_mdz_alignment_length{
+  my $self = shift;
 
+  my $chunks = shift;
+  my $length = 0;
+   for(my $i=0; $i< scalar(@$chunks); $i++){
+     my $chunk = $$chunks[$i];
+     my $type = $self->_get_mdz_chunk_type($chunk);
+
+    if($type eq 'num'){
+      $length = $length + $chunk;
+    }elsif($type eq 'alpha'){
+      $length = $length + length($chunk);
+    }elsif($type eq 'del'){
+      #skip next chunk
+      $i++;
+    }
+
+  }
+  return $length;
+}
+
+=head2 _get_mdz_chunk_type
+
+  Arg [1]    : char
+  Description: get the chunk type
+  Returntype : string
+  Exceptions : none
+  Caller     : internal
+  Status     : Stable
+
+=cut
+
+sub _get_mdz_chunk_type{
+  my $self = shift;
+  my $char = shift;
+
+  my $type;
+  if($char eq '^'){
+    $type = "del";
+  } elsif($char =~ /\d+/){
+    $type = 'num';
+  } elsif($char =~ /\w+/){
+    $type = 'alpha';
+  }
+  return $type;
+}
+
+
+
+=head2 alignment_strings
+
+  Arg [1]    : list of string $flags
+  Example    : $pf->alignment_strings or
+  Description: Allows to rebuild the alignment string of both the seq and hseq sequence
+               using the mdz_string information and the translation object
+  Returntype : array reference containing 2 strings
+               the first corresponds to seq
+               the second corresponds to hseq
+  Exceptions : none
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+
+sub alignment_strings {
+  my $self = shift;
+
+  #Translations
+  my $transl_adaptor = $self->adaptor->db->get_TranslationAdaptor();
+  my $transl_object = $transl_adaptor->fetch_by_dbID($self->translation_id);
```

Comment:
translation_id is in ProteinFeature, not BaseAlignFeature, this needs to be moved to ProteinFeature to prevent someone accidentally calling it and the code going bang.